lib/queue.rb

```diff
@@ -1,31 +1,95 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = []
+    @front = -1
+    @rear = -1
+    @size = 20
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    
+    if (@front == 0 && @rear == @size - 1) #||
+        raise 'An error has occured'
+    elsif (@front == -1) # Queue is empty
+      @front = 0
+      @rear = 0
+      @store[@rear] = element
+    elsif @rear == 19 && @front != 0 # @rear needs to wrap around
+      @rear = 0
+      @store[@rear] = element
+    else
+      @rear = @rear + 1
+      @store[@rear] = element
+    end
```

Comment:
This can work, but you can also use the `rear = (rear + 1) % @store.length` to assign it to the next value.